api/controller.py

```diff
@@ -11,6 +11,7 @@
 import os
 from collections import defaultdict
 
+from expiringdict import ExpiringDict
```

Comment:
An ExpiringDict is a dictionary that gradually forgets about items that were added to it.